p2psrv/server.go

```diff
@@ -47,6 +48,18 @@ func New(opts *Options) *Server {
 		discoveredNodes.Set(node.ID, node)
 	}
 
+	// ConnectOnlyNodes doesn't connect to other sourced nodes
+	connectOnlyNodes := newNodeMap()
+	if len(opts.ConnectOnlyNodes) > 0 {
+		knownNodes = cache.NewPrioCache(5)
+		discoveredNodes = cache.NewRandCache(128)
+		for _, connectOnlyNode := range opts.ConnectOnlyNodes {
+			knownNodes.Set(connectOnlyNode.ID, connectOnlyNode, 0)
+			discoveredNodes.Set(connectOnlyNode.ID, connectOnlyNode)
+			connectOnlyNodes.Add(connectOnlyNode)
+		}
+	}
+
```

Comment:
is there anyway we can test this on unit/integration level ?

Reply:
Yup! Added a test in the New() method. I would like to add more tests, but I believe without refactoring the code it wouldn't be a very efficient test code.